Fixed #10390

TianJinRong opened the pull request:
Closes #10390

Commit:
Optimize the code

Commit:
Revert "Optimize the code"

This reverts commit 24cc9ab00098d7fdd91becb5515a9002b80e55a9.

Commit:
Merge remote-tracking branch 'refs/remotes/yiisoft/master'

Commit:
Fixed #10390

SilverFire commented on framework/helpers/BaseHtml.php:
````diff
@@ -939,9 +940,9 @@ public static function checkboxList($name, $selection = null, $items = [], $opti
      *   corresponding to a single item in $items. The signature of this callback must be:
      *
-     *   ```php
+     *   ~~~
````
```
```php
```
 was correct

TianJinRong replied:
Fixing.

SilverFire commented on framework/helpers/BaseHtml.php:
```diff
@@ -930,6 +930,7 @@ public static function checkboxList($name, $selection = null, $items = [], $opti
      * The following options are specially handled:
      *
      * - tag: string, the tag name of the container element.
+     * - disableTag: boolean, whether to disable the container of the radio buttons. Defaults to false.
```
We don't need this option. It's better to pass `false` to `tag` option

SilverFire commented on framework/helpers/BaseHtml.php:
```diff
@@ -981,10 +982,20 @@ public static function radioList($name, $selection = null, $items = [], $options
+            $tag = '';
+        } else {
+            $tag = isset($options['tag']) ? $options['tag'] : 'div';
+            $tag = json_encode($options['disableTag']);
```
Could you explain the idea of this line?

Commit:
fix bugs

SilverFire commented:
We also need to update PHPUnit tests

SilverFire commented:
Set your email in your local git settings https://help.github.com/articles/setting-your-email-in-git/

SilverFire commented:
We need a CHANGELOG line

TianJinRong commented:
Ok~

Commit:
Fixed #10433

githubjeka commented:
@TianJinRong but you did not do it in the order as is required :) See https://github.com/yiisoft/yii2/blob/master/docs/internals/git-workflow.md#5-update-the-changelog

githubjeka commented on framework/helpers/BaseHtml.php:
```diff
@@ -929,7 +929,8 @@ public static function checkboxList($name, $selection = null, $items = [], $opti
      * The following options are specially handled:
      *
-     * - tag: string, the tag name of the container element.
+     * - tag: string|boolean, the tag name of the container element. Disable the container of the
```
`tag: string|false` only `false` - not `true`. Otherwise misleading.

githubjeka commented on framework/helpers/BaseHtml.php:
```diff
@@ -981,10 +982,19 @@ public static function radioList($name, $selection = null, $items = [], $options
         }
 
-        $tag = isset($options['tag']) ? $options['tag'] : 'div';
+        if (isset($options['tag']) && $options['tag'] == false) {
```
Better `$options['tag'] === false`

githubjeka committed:
Reformat code radioList. A clearer idea.

githubjeka committed:
Update CHANGELOG.md

githubjeka commented:
I sent PR with more clearer the code. https://github.com/TianJinRong/yii2/pull/1/files

githubjeka commented:
https://github.com/yiisoft/yii2/issues/10390

githubjeka committed:
Cosmetic for radioList()

Commit:
Fixed #10433

1. It is readable to get all values of options and unset them at the
beginning of this method;
2. Tag could be set to 'false' if
2. The options would be rebuilt during the transfer before reach this
method, the tag might become 'false' easily. So it would be  better to
accept 'false' and see it as false.
3. ,For tag, it might be confused if we accept one type and a specified
vlaue instead of two types, so there we accept true and see it as
setting default value.

TianJinRong commented:
@githubjeka 
1. false === $tag is great;
2. For tag, it might be confused if we accept one type and a specified
vlaue instead of two types, so there we accept true and see it as
setting default value.

cebe commented:
> For tag, it might be confused if we accept one type and a specified
> value instead of two types, so there we accept true and see it as
> setting default value.

it is a general convention used in Yii to accept some values or bool false to disable it. Introducing true here will be inconsistent.

Commit:
Revert "Fixed #10433"

This reverts commit a6cd1670adcc44278b9ba0f24c7bfcf094c24cbf.

Commit:
Merge pull request #1 from githubjeka/patch-1

Reformat code radioList. A clearer idea.

Commit:
Fixed #10433

Make it readable and clearer.

Commit:
Fixed #10433 Accept only false for tag

Commit:
Merge remote-tracking branch 'refs/remotes/yiisoft/master'

# Conflicts:
#	framework/CHANGELOG.md
#	framework/helpers/BaseHtml.php

TianJinRong commented:
@cebe optimize for the hidden.

SilverFire commented on framework/helpers/BaseHtml.php:
```diff
@@ -975,16 +980,14 @@ public static function radioList($name, $selection = null, $items = [], $options
-            unset($options['unselect']);
-        } else {
-            $hidden = '';
+        if (false === $tag || 'false' === $tag) {
```
We usually write the variable first: `$tag === false`

Also I don't think that `fasle` as a string should do the same as a boolean `false`.

SilverFire commented on framework/helpers/BaseHtml.php:
```diff
@@ -975,16 +980,14 @@ public static function radioList($name, $selection = null, $items = [], $options
         }
 
-        return $hidden . static::tag($tag, implode($separator, $lines), $options);
+        // returned the result with visible content contained by container.
```
I think that the line below is obvious and does not need any comments

SilverFire commented on framework/helpers/BaseHtml.php:
```diff
@@ -953,11 +954,15 @@ public static function checkboxList($name, $selection = null, $items = [], $opti
      */
     public static function radioList($name, $selection = null, $items = [], $options = [])
     {
+        // read options
```
I think that the lines below is obvious and does not need any comments

Commit:
Fixed #10433

Clear the code.

githubjeka commented on framework/helpers/BaseHtml.php:
```diff
@@ -929,7 +929,8 @@ public static function checkboxList($name, $selection = null, $items = [], $opti
      * The following options are specially handled:
      *
-     * - tag: string, the tag name of the container element.
+     * - tag: string|false, the tag name of the container element. Disable the container of the
```
- tag: string|false, string is the tag name of the container element. False to render radio buttons without container.

Commit:
Fixed #10433

githubjeka commented:
:+1:

TianJinRong commented:
@githubjeka  :)

SilverFire commented on framework/helpers/BaseHtml.php:
```diff
@@ -975,16 +979,13 @@ public static function radioList($name, $selection = null, $items = [], $options
-            unset($options['unselect']);
-        } else {
-            $hidden = '';
+        if (false === $tag) {
```
And i'm still voting for `$tag === false`

TianJinRong replied:
Well, i think it doesn`t matter between two scripts if there are no more reasons for deciding which one is better :)

samdark replied:
Yep. There's no sense in doing reverse check in PHP. It's a thing from Java and this way people are avoiding null pointer exceptions.

rugabarbo replied:
This way some developers are avoiding erroneous use `=` instead of `==`|`===`, but I am against this trick.

samdark replied:
It doesn't look natural to me.

TianJinRong replied:
Well, it worth to modify if the script looks wierd.

Commit:
Fixed #10390

SilverFire commented:
Merged. I've also added the same feature to `checkboxList()` method. https://github.com/yiisoft/yii2/commit/85c6a903dced72cce0ce95f8fd424fd492ceb4e6

SilverFire commented:
Thank you!

TianJinRong commented:
@SilverFire :-)
